### DISCUSSION TITLE
Cockpit login fails after installing cpanel

### DISCUSSION THREAD

**Discussion by djsami** [cockpit-project/cockpit#19150] [Q&A] [answered]
### Explain what happens

1. 
2. 
3. 

### Where is the problem in Cockpit?

None

### Server operating system

CentOS

### Server operating system version

Almalinux 8.8

### What browsers are you using?

Chrome

### System log

```shell
Hello

Cockpit was working fine on the installed server. After installing cpanel, it started giving a username and password error, why?

Thansk.
```

**Comment by djsami** [comment]
Username or password is incorrect

**Comment by martinpitt** [comment]
There is no useful information here to diagnose your problem. Did cpanel change/remove user accounts? What exactly do you see in cockpit? Please include screenshot and `sudo journalctl -b` after a failed login attempt, and make sure you can log into the server as that user with SSH .

**Comment by djsami** [comment]
> There is no useful information here to diagnose your problem. Did cpanel change/remove user accounts? What exactly do you see in cockpit? Please include screenshot and `sudo journalctl -b` after a failed login attempt, and make sure you can log into the server as that user with SSH .

Hello

> Jul 29 02:49:35 server.rnmedya.com cockpit-session[55316]: pam_imunify(cockpit:auth): Error connecting to /opt/i360_pam_imunify/pam_imunify360.sock: Permission denied
> Jul 29 02:49:35 server.rnmedya.com unix_chkpwd[55317]: check pass; user unknown
> Jul 29 02:49:35 server.rnmedya.com unix_chkpwd[55318]: check pass; user unknown
> Jul 29 02:49:35 server.rnmedya.com unix_chkpwd[55318]: password check failed for user (root)
> Jul 29 02:49:35 server.rnmedya.com cockpit-session[55316]: pam_unix(cockpit:auth): authentication failure; logname= uid=993 euid=993 tty= ruser= rhost=::ffff:176.42.132.100  user=root
> Jul 29 02:49:37 server.rnmedya.com cockpit-ws[55316]: cockpit-session: open(/var/log/btmp) failed: Permission denied
> Jul 29 02:49:59 server.rnmedya.com sshd[55330]: Accepted password for root from 176.42.132.100 port 41642 ssh2
> Jul 29 02:49:59 server.rnmedya.com systemd[1]: Created slice User Slice of UID 0.
> Jul 29 02:49:59 server.rnmedya.com systemd[1]: Starting User runtime directory /run/user/0...
> Jul 29 02:49:59 server.rnmedya.com systemd-logind[936]: New session 185 of user root.
> Jul 29 02:49:59 server.rnmedya.com systemd[1]: Started User runtime directory /run/user/0.
> Jul 29 02:49:59 server.rnmedya.com systemd[1]: Starting User Manager for UID 0...
> Jul 29 02:49:59 server.rnmedya.com systemd[55335]: pam_unix(systemd-user:session): session opened for user root by (uid=0)
> Jul 29 02:49:59 server.rnmedya.com systemd[55335]: Starting D-Bus User Message Bus Socket.
> Jul 29 02:49:59 server.rnmedya.com systemd[55335]: Reached target Paths.
> Jul 29 02:49:59 server.rnmedya.com systemd[55335]: Reached target Timers.
> Jul 29 02:49:59 server.rnmedya.com systemd[55335]: Listening on Podman API Socket.
> Jul 29 02:49:59 server.rnmedya.com systemd[55335]: Listening on D-Bus User Message Bus Socket.
> Jul 29 02:49:59 server.rnmedya.com systemd[55335]: Reached target Sockets.
> Jul 29 02:49:59 server.rnmedya.com systemd[55335]: Reached target Basic System.
> Jul 29 02:49:59 server.rnmedya.com systemd[55335]: Reached target Default.
> Jul 29 02:49:59 server.rnmedya.com systemd[55335]: Startup finished in 52ms.
> Jul 29 02:49:59 server.rnmedya.com systemd[1]: Started User Manager for UID 0.
> Jul 29 02:49:59 server.rnmedya.com systemd[1]: Started Session 185 of user root.
> Jul 29 02:49:59 server.rnmedya.com sshd[55330]: pam_unix(sshd:session): session opened for user root by (uid=0)
> Jul 29 02:50:00 server.rnmedya.com systemd[1]: Starting system activity accounting tool...
> Jul 29 02:50:00 server.rnmedya.com systemd[1]: sysstat-collect.service: Succeeded.
> Jul 29 02:50:00 server.rnmedya.com systemd[1]: Started system activity accounting tool.
> Jul 29 02:50:01 server.rnmedya.com CROND[55468]: (root) CMD (/usr/sbin/imunify-notifier -update-cron)
> Jul 29 02:50:01 server.rnmedya.com CROND[55470]: (root) CMD (/usr/share/imunify360-webshield/webshield-watchdog &>/dev/null)
> Jul 29 02:50:01 server.rnmedya.com CROND[55471]: (root) CMD (   imunify360-agent malware on-demand check-detached > /dev/null 2>&1  || :)
> Jul 29 02:50:01 server.rnmedya.com CROND[55472]: (root) CMD (   /usr/libexec/imunify360-cgroup-process-wrapper --command /usr/bin/tmpwatch --report-time --report-slice /Imunify.slice/Cron.slice/tmpreaper_tmp_modsec -- 5m /var/imunify360/tmp_modsec)
> Jul 29 02:50:01 server.rnmedya.com systemd[1]: Started Session 188 of user root.
> Jul 29 02:50:01 server.rnmedya.com systemd[1]: Started Session 187 of user root.
> Jul 29 02:50:01 server.rnmedya.com CROND[55488]: (root) CMD (/usr/local/cpanel/scripts/dcpumon-wrapper >/dev/null 2>&1)
> Jul 29 02:50:01 server.rnmedya.com CROND[55489]: (root) CMD (/usr/local/cpanel/scripts/eximstats_spam_check 2>&1)
> Jul 29 02:50:01 server.rnmedya.com systemd[1]: session-187.scope: Succeeded.
> Jul 29 02:50:04 server.rnmedya.com imunify360-php-daemon[52948]: error sending perf stats: circuit breaker is open
> Jul 29 02:50:04 server.rnmedya.com pure-ftpd[55597]: (?@198.235.24.224) [INFO] New connection from 198.235.24.224
> Jul 29 02:50:04 server.rnmedya.com systemd[1]: session-188.scope: Succeeded.
> Jul 29 02:50:05 server.rnmedya.com pure-ftpd[55597]: (?@198.235.24.224) [INFO] TLS: Enabled TLSv1.2 with ECDHE-RSA-AES256-GCM-SHA384, 256 secret bits cipher
> Jul 29 02:50:05 server.rnmedya.com pure-ftpd[55597]: (?@198.235.24.224) [INFO] Logout.
> Jul 29 02:50:37 server.rnmedya.com cockpit-tls[55277]: cockpit-tls: gnutls_handshake failed: A TLS fatal alert has been received.
> Jul 29 02:50:37 server.rnmedya.com cockpit-session[55624]: pam_imunify(cockpit:auth): Error connecting to /opt/i360_pam_imunify/pam_imunify360.sock: Permission denied
> Jul 29 02:50:37 server.rnmedya.com unix_chkpwd[55625]: check pass; user unknown
> Jul 29 02:50:37 server.rnmedya.com unix_chkpwd[55626]: check pass; user unknown
> Jul 29 02:50:37 server.rnmedya.com unix_chkpwd[55626]: password check failed for user (root)
> Jul 29 02:50:37 server.rnmedya.com cockpit-session[55624]: pam_unix(cockpit:auth): authentication failure; logname= uid=993 euid=993 tty= ruser= rhost=::ffff:176.42.132.100  user=root
> Jul 29 02:50:39 server.rnmedya.com cockpit-ws[55624]: cockpit-session: open(/var/log/btmp) failed: Permission denied
> Jul 29 02:50:39 server.rnmedya.com systemd[1]: cockpit-wsinstance-https@e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.service: Succeeded.
> Jul 29 02:50:39 server.rnmedya.com systemd[1]: cockpit-wsinstance-https@e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.socket: Succeeded.
> Jul 29 02:50:39 server.rnmedya.com systemd[1]: Closed Socket for Cockpit Web Service https instance e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.
> Jul 29 02:51:01 server.rnmedya.com CROND[55649]: (root) CMD (/usr/sbin/imunify-notifier -update-cron)
> 


![Screenshot_2](https://github.com/cockpit-project/cockpit/assets/36164830/1bda2a99-6267-46ac-9ee6-b6ef27144aff)

**Comment by martinpitt** [comment, marked as answer]
> pam_imunify(cockpit:auth): Error connecting to /opt/i360_pam_imunify/pam_imunify360.sock: Permission denied
>unix_chkpwd[55625]: check pass; user unknown
> server.rnmedya.com unix_chkpwd[55626]: password check failed for user (root)

Seems either cpanel or you changed the PAM config so that it is broken now? You can also try to log in as a non-root user, but that pam_immunify error doesn't look healthy either :cry:
